sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/eventlogger/EventLogger.kt

```diff
@@ -62,4 +96,125 @@ object EventLogger {
     ) {
         // Intentionally left blank. Will be supported later.
     }
+
+    // ------------------------------------Internal APIs-----------------------------------------------
+
+    private fun buildEventLoggerHttpClient(baseUrl: String): Retrofit {
+        return Retrofit
+            .Builder()
+            .baseUrl(baseUrl)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+    }
+
+    private fun initialize(
+        context: Context,
+        apiUrl: String,
+        apiKey: String
+    ) {
+        // Build dependencies
+        initialize(
+            eventsSender = RetrofitEventsSender(
+                buildEventLoggerHttpClient(apiUrl).create(RetrofitEventsSender.Api::class.java),
+                apiKey
+            ),
+            eventsStorage = SharedPreferencesEventsStorage(
+                context.getSharedPreferences(Config.EVENTS_STORAGE_FILENAME, Context.MODE_PRIVATE)
+            ),
+            eventLoggerCache = SharedPreferencesEventLoggerCache(
+                context.getSharedPreferences(Config.EVENT_LOGGER_GENERAL_CACHE_FILENAME, Context.MODE_PRIVATE)
+            ),
+            eventLoggerHelper = EventLoggerHelper(
+                WeakReference(context.applicationContext)
+            ),
+            tasksQueue = Executors.newSingleThreadExecutor()
+        )
+    }
+
+    /**
+     * Registers to the background to foreground app transition and if the TTL expired, will send all events in storage.
+     */
+    @VisibleForTesting
+    @Synchronized
+    internal fun initialize(
+        eventsSender: EventsSender,
+        eventsStorage: EventsStorage,
+        eventLoggerCache: EventLoggerCache,
+        eventLoggerHelper: EventLoggerHelper,
+        tasksQueue: ExecutorService
+    ) {
+        this.eventsSender = eventsSender
+        this.eventsStorage = eventsStorage
+        this.eventLoggerCache = eventLoggerCache
+        this.eventLoggerHelper = eventLoggerHelper
+        this.tasksQueue = tasksQueue
+        this.isConfigureCalled = true
+
+        tasksQueue.safeExecute {
+            registerToAppTransitions()
+            if (isTtlExpired()) {
+                sendAllEventsInStorage()
+            }
+        }
+    }
+
+    private fun registerToAppTransitions() {
+        // ToDo
+    }
+
+    private fun sendAllEventsInStorage() {
+        val allEvents = eventsStorage.getAllEvents()
+        sendEvents(
+            events = allEvents.values.toList(),
+            onSuccess = {
+                eventLoggerCache.setTtlReferenceTime(System.currentTimeMillis())
```

Comment:
is this to record when was the last time event were sent to server successfully ?

Reply:
Yes, also if never pushed before, this same method is used to set the reference time in line 191. Hence, the naming.